src/meta/processors/admin/Balancer.cpp

```diff
@@ -192,28 +192,36 @@ bool Balancer::getAllSpaces(std::vector<std::tuple<GraphSpaceID, int32_t, bool>>
     return true;
 }
 
-cpp2::ErrorCode Balancer::buildBalancePlan(std::unordered_set<HostAddr> hostDel) {
-    CHECK(!plan_) << "plan should be nullptr now";
+cpp2::ErrorCode Balancer::buildBalancePlan(std::vector<HostAddr>&& lostHosts) {
+    if (plan_ != nullptr) {
+        LOG(ERROR) << "Balance plan should be nullptr now";
+        return cpp2::ErrorCode::E_CONFLICT;
+    }
+
     std::vector<std::tuple<GraphSpaceID, int32_t, bool>> spaces;
     if (!getAllSpaces(spaces)) {
         LOG(ERROR) << "Can't get all spaces";
         return cpp2::ErrorCode::E_STORE_FAILURE;
     }
+
     plan_ = std::make_unique<BalancePlan>(time::WallClock::fastNowInSec(), kv_, client_);
-    for (auto spaceInfo : spaces) {
-        auto taskRet = genTasks(std::get<0>(spaceInfo),
-                                std::get<1>(spaceInfo),
-                                std::get<2>(spaceInfo),
-                                hostDel);
+    for (const auto& spaceInfo : spaces) {
+        auto spaceId = std::get<0>(spaceInfo);
```

Comment:
I suggest move the genTasks and related logic to JobExecutor in this PR. We don't support balance all space anymore later

Reply:
I  will move to job manager later because some modify need to check carefully

Reply:
👌